enable support for decoupled in-car settings

mcowger opened the pull request:
Adds a toggle for decoupling the in car option position from the pit option enablement.  Defaults to "coupled" to match previous behavior.  

When disabled, it allows pit menu actions even if the in car rotary is not set to "put". 

(also fixed a spelling error and some build options).  

Follow up PR coming to change the InCar menu "pit" option to "-" when coupling is disabled.  

Fixes #1

mcowger commented:
Attached update that does the pit menu change described above.
[DahlDesignDDU+NewVersionCheck+PitEnforceFix.zip](https://github.com/andreasdahl1987/DahlDesignProperties/files/9912193/DahlDesignDDU%2BNewVersionCheck%2BPitEnforceFix.zip)

andreasdahl1987 commented:
Excellent, it works great :)

Something happened here:
![image](https://user-images.githubusercontent.com/40788634/199356655-91cc55a7-03fc-4b95-877d-f9e48687f982.png)

Looks like you didnt remove anything, what might have happened?

mcowger commented:
Oh weird.  Will play with it tonight to see what happened.

On Tue, Nov 1, 2022 at 3:50 PM Andreas Dahl ***@***.***>
wrote:

> Excellent, it works great :)
>
> Something happened here:
> [image: image]
> <https://user-images.githubusercontent.com/40788634/199356655-91cc55a7-03fc-4b95-877d-f9e48687f982.png>
>
> Looks like you didnt remove anything, what might have happened?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/andreasdahl1987/DahlDesignProperties/pull/5#issuecomment-1299324519>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAOXCTHJ3JDBUATPC34GLKTWGGNCVANCNFSM6AAAAAARTU5WUY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
-- Matt

mcowger commented:
Fixed the formatting issue.  Cause was that the `TextBox` and `SHToggle` items were outside the stack panel (which is why you'd had to set weird margin offsets), and that was impacted when I changed the max window height to 1080.  I put them inside the stack panel and removed the margins and its all good.

mcowger commented on SettingsControlDemo.xaml:
```diff
@@ -314,6 +292,13 @@
                                 Height="20"
                                 IsChecked="{Binding LapInfoScreen}" Margin="14,-20,0,10"/>
+                            <TextBlock
+                                Text="Couple Pit Controls Enable to In-Car Rotary - Support mode fual dual 12-position switches" Margin="10,0,0,23"/>
```
I dont think this is quite right (besides the typo fix :)).

The intent of this change is to allow pitmenu controls to function even if the in-car rotary is not at position 12 (Pit), not to share the buttons (I'm not actually doing that). 

How about "Allow Pit Controls regardless of In Car Rotary Position"

andreasdahl1987 replied:
> I dont think this is quite right (besides the typo fix :)).
> 
> The intent of this change is to allow pitmenu controls to function even if the in-car rotary is not at position 12 (Pit), not to share the buttons (I'm not actually doing that).
> 
> How about "Allow Pit Controls regardless of In Car Rotary Position"

That's a better phrasing. However, when `true`, the Pit menu controls are locked untill inCar is 12, right? So we'd have to flip it?

mcowger replied:
You are absolutely right.  Thats what I get for coding before caffeine :).

"Disable Pit Controls unless In-Car Rotary on Pit Setting"?

andreasdahl1987 replied:
> You are absolutely right. Thats what I get for coding before caffeine :).
> 
> "Disable Pit Controls unless In-Car Rotary on Pit Setting"?

How about "Disable PitMenu Controls unless In-Car Rotary unused or in 12th position"?

mcowger replied:
Fixed in new commit.
